File: banchus-common/src/main/java/pe/nanamochi/banchus/domain/enums/BeatmapDirectDisplayMode.java
package pe.nanamochi.banchus.domain.enums;

import lombok.AllArgsConstructor;
import lombok.Getter;

@Getter
@AllArgsConstructor
public enum BeatmapDirectDisplayMode {
  RANKED(0, 1),
  RANKED_STRICT(1, 1),
  PENDING(2, 0),
  QUALIFIED(3, 3),
  ALL(4, -1),
  GRAVEYARD(5, -2),
  APPROVED(6, 2),
  RANKED_PLAYED(7, 1),
  LOVED(8, 4);

  private final int value;
  private final int apiStatus;

  public static BeatmapDirectDisplayMode fromValue(int value) {
    for (BeatmapDirectDisplayMode mode : values()) {
      if (mode.value == value) return mode;
    }
    return ALL;
  }
}

File: banchus-common/src/main/kotlin/pe/nanamochi/banchus/domain/enums/BeatmapDirectDisplayMode.kt
package pe.nanamochi.banchus.domain.enums

enum class BeatmapDirectDisplayMode(val value: Int, val apiStatus: Int) {
    RANKED(0, 1),
    RANKED_STRICT(1, 1),
    PENDING(2, 0),
    QUALIFIED(3, 3),
    ALL(4, -1),
    GRAVEYARD(5, -2),
    APPROVED(6, 2),
    RANKED_PLAYED(7, 1),
    LOVED(8, 4);

    companion object {
        fun fromValue(value: Int): BeatmapDirectDisplayMode {
            return entries.find { it.value == value } ?: ALL
        }
    }
}
